function-sidecar/pkg/dispatcher/grpc/grpc.go

```diff
@@ -114,14 +177,23 @@ func NewGrpcDispatcher(port int, timeout time.Duration) (dispatcher.Dispatcher,
 		return nil, err
 	}
 
-	fnStream, err := function.NewMessageFunctionClient(conn).Call(context.Background())
-	if err != nil {
-		return nil, err
+	result := &grpcDispatcher{
+		client:         function.NewMessageFunctionClient(conn),
+		input:          make(chan message.Message, 100),
+		output:         make(chan message.Message, 100),
+		closed:         make(chan struct{}),
+		streams:        make(map[interface{}]ongoingStream),
+		windowingFactory: unmarshallFactory(),
+		correlation: func(m message.Message) interface{} {
+			// Correlate by correlationId header by default if present
+			if c, ok := m.Headers()[correlationId] ; ok {
```

Comment:
As explained in the `CorrelationStrategy` type doc, this is what I call the "spatial" correlation (similar to say, the grouping correlation strategy that an SI aggregator does).

I'm using the `correlationId` here by default because the sidecar is now responsible for forwarding the `correlationId` on the way back. If we did not correlate (ie if there was just 1 giant group, even though of size 1) it is possible that response for request B ends up as the response for request A

Reply:
I don't think we're ready to get into aggregation use-cases yet.

Reply:
Now that I think of it, this should not even be necessary as if a window is already opened (waiting for its response) and a new request comes in, a new window is created and replaces the current one in the map (assuming no correlation whatsoever). But the evicted window should still "live" and wait for the reply (with some eventual reaping mechanism)

Reply:
Well, my (unconcious at the time) reasoning is that if we move the windowing out of the function, potential correlation has to come with it, because I think in most cases you want correlation to happen _before_ windowing

Reply:
You are referring to what would normally go into a `groupBy(..)` on a `Flux` for example? That could happen before or after windowing, depending on the use-case. I think we should be careful not to conflate "correlation" with "grouping", at least not yet (we have a very specific role for correlation today, and it's not about grouping for processing).